python/cudf/cudf/core/index.py

```diff
@@ -435,6 +439,123 @@ def _concat(cls, objs):
         result.name = name
         return result
 
+    def append(self, other):
+        """
+        Append a collection of Index options together.
+
+        Parameters
+        ----------
+        other : Index or list/tuple of indices
+
+        Returns
+        -------
+        appended : Index
+
+        Examples
+        --------
+        >>> import cudf
+        >>> idx = cudf.Index([1, 2, 10, 100])
+        >>> idx
+        Int64Index([1, 2, 10, 100], dtype='int64')
+        >>> other = cudf.Index([200, 400, 50])
+        >>> other
+        Int64Index([200, 400, 50], dtype='int64')
+        >>> idx.append(other)
+        Int64Index([1, 2, 10, 100, 200, 400, 50], dtype='int64')
+
+        append accepts list of Index objects
+
+        >>> idx.append([other, other])
+        Int64Index([1, 2, 10, 100, 200, 400, 50, 200, 400, 50], dtype='int64')
+        """
+
+        if isinstance(other, (list, tuple)):
+            to_concat = [self] + list(other)
+        else:
+
+            this = self
+            if len(other) == 0:
+                # short-circuit and return a copy
+                to_concat = [self]
+            if len(self) == 0:
+                to_concat = [other]
+
+            if len(self) and len(other):
+                if (this.dtype == "object" and other.dtype != "object") or (
+                    other.dtype == "object" and this.dtype != "object"
+                ):
```

Comment:
I think we need to validate / convert `other` to an Index before we hit here. I.E. move the check for all inputs being an Index up above here.